lib/api/core/httpRouter/index.js

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+var
+  RoutePart = require('./routePart'),
+  InternalError = require('kuzzle-common-objects').Errors.internalError,
+  BadRequestError = require('kuzzle-common-objects').Errors.badRequestError,
+  NotFoundError = require('kuzzle-common-objects').Errors.notFoundError,
+  RequestObject = require('kuzzle-common-objects').Models.requestObject,
+  ResponseObject = require('kuzzle-common-objects').Models.responseObject,
+  HttpResponse = require('../entryPoints/httpResponse'),
+  URL = require('url');
+
+/**
+ * Attach handler to routes and dispatch a HTTP
+ * message to the right handler
+ *
+ * Handlers will be called with the following arguments:
+ *   - request: received HTTP request
+ *   - response: HTTP response object
+ *   - data: URL query arguments and/or POST data, if any
+ *
+ * @constructor
+ */
+function Router() {
+  this.routes = {
+    GET: new RoutePart(),
+    POST: new RoutePart(),
+    PUT: new RoutePart(),
+    DELETE: new RoutePart()
+  };
+
+  return this;
+}
+
+/**
+ * Attach a handler to a GET HTTP route
+ *
+ * @param {string} url
+ * @param {Function} handler
+ */
+Router.prototype.get = function routerGet(url, handler) {
+  attach(url, handler, this.routes.GET);
+};
+
+/**
+ * Attach a handler to a POST HTTP route
+ *
+ * @param {string} url
+ * @param {Function} handler
+ */
+Router.prototype.post = function routerGet(url, handler) {
+  attach(url, handler, this.routes.POST);
+};
+
+/**
+ * Attach a handler to a PUT HTTP route
+ *
+ * @param {string} url
+ * @param {Function} handler
+ */
+Router.prototype.put = function routerGet(url, handler) {
+  attach(url, handler, this.routes.PUT);
+};
+
+/**
+ * Attach a handler to a DELETE HTTP route
+ *
+ * @param {string} url
+ * @param {Function} handler
+ */
+Router.prototype.delete = function routerGet(url, handler) {
+  attach(url, handler, this.routes.DELETE);
+};
+
+/**
+ * Route an incoming HTTP request to the right handler
+ *
+ * @param {object} request - HTTP request formatted by Kuzzle Proxy
+ * @param {function} cb
+ */
+Router.prototype.route = function route (request, cb) {
+  let
+    parsed = URL.parse(request.url, true),
+    base = {
+      requestId: request.requestId,
+      headers: request.headers
+    },
+    handler,
+    requestObject;
+
+  if (!this.routes[request.method]) {
+    requestObject = new RequestObject(base, parsed.query, 'rest');
+    return replyWithError(cb, requestObject, new BadRequestError(`Unrecognized HTTP method ${request.method}`));
+  }
+
+  handler = this.routes[request.method].getHandler(parsed.pathname.split('/'), parsed.query);
+
+  // Do not duplicate those informations to the data.body part of the request object
```

Comment:
That work was previously performed by `RouterController.executeFromRest`, where we analyzed the request params and picked the index/collection only once. I wanted something less tedious, but I ended up with this, which is almost exactly what was done before :-(